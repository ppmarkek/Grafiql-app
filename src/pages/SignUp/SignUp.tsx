import { Button, Grid, List, ListItem, Paper } from '@mui/material';
import { FormikProps, useFormik } from 'formik';
import FormInput, { Values } from '../../components/FormInput/FormInput';

import { useFirebaseAuth } from '../../services/auth/firebase';
import { Link } from 'react-router-dom';
import GoogleButton from '../../components/GoogleButton/GoogleButton';
import { useI18n } from '../../components/Context/ValueContext';

const SignUp = () => {
  const i18n = useI18n();

  const {
    user,
    loading,
    error,
    registerWithEmailAndPassword,
    signInWithGoogle,
  } = useFirebaseAuth();
  console.log(user, loading, error);
  const formik: FormikProps<Values> = useFormik({
    initialValues: {
      name: '',
      email: '',
      password: '',
      passwordConfirm: '',
    },
    onSubmit: (values) => {
      registerWithEmailAndPassword(values.name, values.email, values.password);
      alert(JSON.stringify(values, null, 2));
    },
  });
  return (
    <Grid
      sx={{
        width: '100%',
        height: '100%',
        position: 'relative',
        display: 'flex',
        alignItems: 'center',
      }}
    >
      <Paper
        elevation={3}
        sx={{
          width: '60vw',
          height: '60vh',
          display: 'flex',
          flexDirection: 'column',
          justifyContent: 'space-around',
          alignItems: 'center',
          position: 'relative',
          margin: 'auto',
        }}
      >
        <h2>{i18n.auth.signUp}</h2>
        <GoogleButton
          googleButtonName={i18n.auth.googleButton}
          signInWithGoogleHandler={signInWithGoogle}
        />
        <form onSubmit={formik.handleSubmit}>
          <List
            sx={{ width: '100%', maxWidth: 360, bgcolor: 'background.paper' }}
          >
            <ListItem>
              <FormInput
                title={'name'}
                text={i18n.auth.name}
                type={'text'}
                formik={formik}
              />
            </ListItem>
            <ListItem>
              <FormInput
                title={'email'}
                text={i18n.auth.email}
                type={'email'}
                formik={formik}
              />
            </ListItem>
            <ListItem>
              <FormInput
                title={'password'}
                text={i18n.auth.password}
                type={'password'}
                formik={formik}
              />
            </ListItem>
            <ListItem>
              <FormInput
                title={'passwordConfirm'}
                text={i18n.auth.passwordConfirm}
                type={'password'}
                formik={formik}
              />
            </ListItem>
            <ListItem>
              <Button type="submit">{i18n.auth.buttonName}</Button>
            </ListItem>
            <ListItem>
              <Grid sx={{ marginRight: '1rem' }}>{i18n.auth.haveAccount}</Grid>
              <Link to={'/signin'}>{i18n.auth.signInLink}</Link>
            </ListItem>
          </List>
        </form>
      </Paper>
    </Grid>
  );
};

export default SignUp;
