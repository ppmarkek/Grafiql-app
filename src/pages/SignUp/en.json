{
	"auth": {
		"signIn": "Welcome back to GraphQL Proxy!",
		"signUp": "New User Registration",
		"name": "name",
		"email": "email",
		"password": "password",
		"passwordConfirm": "password confirmation",
		"buttonName": "Submit",
		"googleButton": "Continue with Google",
		"dontHaveAccount": "Don't have account?",
		"haveAccount": "Already have an account?",
		"signInLink": "SignIn",
		"signUpLink": "SignUp"
	}
}
