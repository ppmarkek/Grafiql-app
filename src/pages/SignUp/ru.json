{
	"auth": {
		"signIn": "Добро пожаловать на GraphQL Proxy!",
		"signUp": "Регистрация нового пользователя",
		"name": "Имя",
		"email": "Электронная почта",
		"password": "Пароль",
		"passwordConfirm": "Подтверждение пароля",
		"buttonName": "Отправить",
		"googleButton": "Продолжить с Google",
		"dontHaveAccount": "Нет учетной записи?",
		"haveAccount": "Есть учетная запись?",
		"signInLink": "Вход",
		"signUpLink": "Регистрация"
	}
}
