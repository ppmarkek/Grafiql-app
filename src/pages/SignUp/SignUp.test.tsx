import React from 'react';
import SignUp from './SignUp';
import renderer from 'react-test-renderer';

describe('SignUp', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });
  it('should render without errors', () => {
    const tree = renderer.create(<SignUp />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
