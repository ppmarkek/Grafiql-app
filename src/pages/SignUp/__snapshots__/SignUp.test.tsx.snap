// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should render without errors 1`] = `
<div
  className="MuiGrid-root css-x9vj63-MuiGrid-root"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-dfu1h7-MuiPaper-root"
  >
    <h1>
      Регистрация
    </h1>
    <form
      onSubmit={[Function]}
    >
      <ul
        className="MuiList-root MuiList-padding css-1mcnwpj-MuiList-root"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          disabled={false}
        >
          <div
            className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="name"
            >
              Имя
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-describedby="component-error-text"
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="name"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Имя
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </li>
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          disabled={false}
        >
          <div
            className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="email"
            >
              Электронная почта
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-describedby="component-error-text"
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="email"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Электронная почта
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </li>
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          disabled={false}
        >
          <div
            className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="password"
            >
              Пароль
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-describedby="component-error-text"
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Пароль
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </li>
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          disabled={false}
        >
          <div
            className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="passwordConfirm"
            >
              Подтверждение пароля
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-describedby="component-error-text"
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="passwordConfirm"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Подтверждение пароля
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </li>
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          disabled={false}
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            Отправить
          </button>
        </li>
      </ul>
    </form>
  </div>
</div>
`;
